include/deal.II/lac/trilinos_tpetra_vector.templates.h

```diff
@@ -90,12 +111,20 @@ namespace LinearAlgebra
                            const MPI_Comm  communicator,
                            const bool      omit_zeroing_entries)
     {
-      Teuchos::RCP<MapType> input_map =
+      // release memory before reallocation
+      vector.reset();
+      locally_owned_map.reset();
+      locally_relevant_map.reset();
+
+      // here we assume, that the index set is non-overlapping
+      compressed = true;
```

Comment:
Can we assert that at some point in the function?

Reply:
In the next line is the function:
```
locally_owned_map = 
  parallel_partitioner.make_tpetra_map_rcp(communicator, false);
```
Which will throw an Assert if the index set `parallel_partitioner` is overlapping.

I  changed the comment so it is clear that `make_tpetra_map_rcp()` will check if the index set `parallel_partitioner` is overlapping:
```
// Here we assume, that the index set is non-overlapping. 
// If the index set is overlapping the function 
// make_tpetra_map_rcp() will throw an assert.
```